rust/operator/src/container.rs

```diff
@@ -0,0 +1,778 @@
+        }) = container_log_config
+        {
+            product_logging::framework::capture_shell_output(
+                STACKABLE_LOG_DIR,
+                container_name,
+                &log_config,
+            )
+        } else {
+            String::new()
+        }
+    }
+
+    /// Return volume mount directories depending on the container.
+    fn volume_mount_dirs(&self) -> &ContainerVolumeDirs {
+        match &self {
+            ContainerConfig::Hdfs { volume_mounts, .. } => volume_mounts,
+            ContainerConfig::Zkfc { volume_mounts, .. } => volume_mounts,
+            ContainerConfig::FormatNameNodes { volume_mounts, .. } => volume_mounts,
+            ContainerConfig::FormatZooKeeper { volume_mounts, .. } => volume_mounts,
+            ContainerConfig::WaitForNameNodes { volume_mounts, .. } => volume_mounts,
+        }
+    }
+
+    /// Returns the container command.
+    fn command(&self) -> Vec<String> {
+        match self {
+            ContainerConfig::Hdfs { .. } | ContainerConfig::Zkfc { .. } => vec![
+                "/bin/bash".to_string(),
+                "-x".to_string(),
+                "-euo".to_string(),
+                "pipefail".to_string(),
+                "-c".to_string(),
+            ],
+            ContainerConfig::FormatNameNodes { .. }
+            | ContainerConfig::FormatZooKeeper { .. }
+            | ContainerConfig::WaitForNameNodes { .. } => {
+                vec!["bash".to_string(), "-c".to_string()]
+            }
+        }
+    }
+
+    /// Returns the container command arguments.
+    fn args(
+        &self,
+        merged_config: &(dyn MergedConfig + Send + 'static),
+        namenode_podrefs: &[HdfsPodRef],
+    ) -> Vec<String> {
+        let mut args = vec![];
+
+        match self {
+            ContainerConfig::Hdfs { role, .. } => {
+                args.push(self.create_config_directory_cmd());
+                args.push(self.copy_hdfs_and_core_site_xml_cmd());
+                args.push(self.copy_log4j_properties_cmd(
+                    HDFS_LOG4J_CONFIG_FILE,
+                    merged_config.hdfs_logging(),
+                ));
+
+                args.push(format!(
+                    "{hadoop_home}/bin/hdfs --debug {role}",
+                    hadoop_home = Self::HADOOP_HOME,
+                ));
+            }
+            ContainerConfig::Zkfc { .. } => {
+                args.push(self.create_config_directory_cmd());
+                args.push(self.copy_hdfs_and_core_site_xml_cmd());
+                if let Some(container_config) = merged_config.zkfc_logging() {
+                    args.push(
+                        self.copy_log4j_properties_cmd(ZKFC_LOG4J_CONFIG_FILE, container_config),
+                    );
+
+                    args.push(format!(
+                        "{hadoop_home}/bin/hdfs zkfc",
+                        hadoop_home = Self::HADOOP_HOME
+                    ));
+                }
```

Comment:
The command should probably be independent of the logging configuration:

```suggestion
                }

                args.push(format!(
                    "{hadoop_home}/bin/hdfs zkfc",
                    hadoop_home = Self::HADOOP_HOME
                ));
```

Reply:
good point!